internal/subscriber/ordering_test.go

```diff
@@ -98,6 +99,87 @@ func TestOrderedImplementation_ModAckDeadline(t *testing.T) {
 	assert.Zero(t, len(subImpl.consumedMessageStats[tp].consumedMessages))
 }
 
+func TestOrderedImplementation_GetConsumerLag(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	cs := getMockConsumer(ctx, ctrl)
+	consumer := getMockConsumerManager(ctx, ctrl, cs)
+	offsetRepo := getMockOffsetRepo(ctrl, string(sequenceSuccess))
+	subImpl := getMockOrderedImplementation(ctx, consumer, offsetRepo)
+	pTopic := subImpl.topic
+	rTopic := subImpl.subscription.GetRetryTopic()
+	tests := []struct {
+		name string
+		want map[string]uint64
+	}{
+		{
+			name: "Get Consumer Lag for Ordered Implementation",
+			want: map[string]uint64{
+				pTopic: 0,
+				rTopic: 0,
+			},
+		},
+		{
+			name: "Get Positive Consumer Lag for Ordered Implementation",
+			want: map[string]uint64{
+				pTopic: 1,
+				rTopic: 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cs.EXPECT().FetchConsumerLag(gomock.Any()).Return(tt.want, nil).Times(1)
+			got := subImpl.GetConsumerLag()
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("BasicImplementation.GetConsumerLag() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestOrderedImplementation_EvictUnackedMessagesPastDeadline(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+	cs := getMockConsumer(ctx, ctrl)
+	consumer := getMockConsumerManager(ctx, ctrl, cs)
+	offsetRepo := getMockOffsetRepo(ctrl, string(sequenceSuccess))
+	subImpl := getMockOrderedImplementation(ctx, consumer, offsetRepo)
+	orderedConsumptionMetadata := NewOrderedConsumptionMetadata()
+	orderedConsumptionMetadata.Store(getReceivedMessage(), time.Now().Add(time.Second*(-1)).Unix())
```

Comment:
this seems like something that is internal to subscriber implementation of evicting messages. I think we should refrain from modifying its values. Better test case would be to pull a new message with a pre-defined deadline, and then evicting messages.

Reply:
I didn't get you, please explain